### PR TITLE
`Paywalls`: implemented `LoadingPaywallView` with `placeholder`

### PR DESCRIPTION
**Pull request by NachoSoto**
This uses https://github.com/google/accompanist/blob/main/placeholder/. It's deprecated but might come to Compose itself in the future. This is just a temporary loading screen, we can decide whether to keep it, implement it in another way, or remove it in the future.

image

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt [hunk 1]
```diff
@@ -62,7 +69,11 @@ internal fun Template2(state: PaywallViewState.Loaded, viewModel: PaywallViewMod
 }
 
 @Composable
-private fun Template2MainContent(state: PaywallViewState.Loaded, viewModel: PaywallViewModel) {
+private fun Template2MainContent(
+    state: PaywallViewState.Loaded,
+    viewModel: PaywallViewModel,
+    childModifier: Modifier,
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt:75]: I wonder if we should rename to `contentModifier` or `loadingContentModifier` so it's clearer.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt [hunk 2]
```diff
@@ -126,7 +140,7 @@ private fun SelectPackageButton(
     }
     val border = if (isSelected) null else BorderStroke(2.dp, colors.text1)
     Button(
-        modifier = Modifier.fillMaxWidth(),
+        modifier = Modifier.fillMaxWidth().then(childModifier),
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt:143]: You could do 
```suggestion
        modifier = childModifier.fillMaxWidth(),
```
I think? We should see how it looks though.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt [hunk 3]
```diff
@@ -29,13 +34,23 @@ internal fun LoadingPaywallView(mode: PaywallViewMode) {
     val provider = VariableDataProvider(LocalContext.current.applicationContext.toAndroidContext())
     val offering = LoadingPaywallConstants.offering
 
+    val placeholder = Modifier.placeholder(
+        visible = true,
+        highlight = com.revenuecat.purchases.ui.revenuecatui.composables.Fade(
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt:39]: We should try to use a normal import

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt (hunk 1 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt:75]: 👍🏻

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt (hunk 2 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt:143]: Yeah that's simpler 👍🏻

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt (hunk 3 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt:39]: Oopss thanks

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/placeholder.kt [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+        infiniteRepeatable(
+            animation = tween(durationMillis = 1700, delayMillis = 200),
+            repeatMode = RepeatMode.Restart,
+        )
+    }
+}
+
+/**
+ * From https://github.com/google/accompanist/blob/main/placeholder/src/main/java/com/google/accompanist/placeholder/Placeholder.kt
+ * See https://github.com/google/accompanist/pull/1672. That's deprecated, so we're copying it here for now.
```
Comment (tonidero) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/placeholder.kt:66]: 👍

**Review by tonidero — APPROVED** [review]
I'd love to get this in if it works well. We should make sure it works well on different devices. But I think the code looks good for now.

**Pull request merged**